### PR TITLE
Add docker_network options

### PR DESCRIPTION
**Pull request by marchenm**
This pr allows docker_networks to be specified without containers, but also has backwards compatibility.

Also adds advanced network options.

### CODE REVIEW

**Comment by senorsmile** [comment]
It looks good to me.  Have you tested it on real infra yet?  The "when: " clauses may need to be modified if they don't work as is.  I'd like to see end results of when docker_networks.containers is defined, and also when it is not.

**Comment by marchenm** [comment]
I've fixed the when statements and tested both cases.

File: tasks/main.yml [hunk 1]
```diff
@@ -30,11 +30,16 @@
       docker_app_num: 0
     tags: docker_start
 
+  - include: docker_network.yml
+    when: docker_networks | selectattr('containers', 'undefined')
```
Comment (senorsmile) [inline tasks/main.yml]: I see what you're doing there!  

The easier way to specify this (and safer) would be:
```
when:
  - docker_networks is defined
  - docker_networks.containers is defined
```

the list passed to when is a logical AND

File: tasks/main.yml [hunk 2]
```diff
@@ -30,11 +30,16 @@
...
+    tags: docker_network
+
   - include: docker_start.yml
     with_items: "{{ docker_apps }}"
     tags: docker_start
 
   - include: docker_network.yml
+    when: docker_networks | selectattr('containers', 'defined')
```
Comment (senorsmile) [inline tasks/main.yml]: and for this one
```
when: ( 
    docker_networks is undefined
    or
    docker_networks.containers is undefined
  )
```

**Review by senorsmile — CHANGES_REQUESTED** [review]

**Comment by marchenm** [comment]
Fixed. I've tested both cases and it works.

**Review by senorsmile — APPROVED** [review]
lgtm :+1:

**Pull request merged**